File: Matlab code/pressureDrivenCurrent.m

% Inlet Trondheimsfjorden: 176, 56


% Enter correct path to data set:
filename = 'E:/nn9828k/spring2021/mids_short/dataset.nc';

% Open file:
ncid = netcdf.open(filename);

pos1 = [176 53];
pos2 = [176 56];
pos3 = [176 59];
% pos1 = [200 247];
% pos2 = [200 250];
% pos3 = [200 253];


[t, E_inner] = getTimeSeries(ncid, 'elevation', pos1, []);
[t, E_outer] = getTimeSeries(ncid, 'elevation', pos3, []);

[t, V] = getTimeSeries(ncid, 'v-velocity', pos2, 10);
[t, U] = getTimeSeries(ncid, 'u-velocity', pos2, 10);
[t, T] = getTimeSeries(ncid, 'temperature', pos2, 10);
[t, S] = getTimeSeries(ncid, 'salinity', pos2, 10);
[t, windV] = getTimeSeries(ncid, 'v-Wind', pos2, []);

V_acc = [0; diff(V)]/(86400*(t(2)-t(1)));


% Close file:
netcdf.close(ncid);

%%
g = 9.81;
rho_0 = 1023.6; % Density of seawater

% Find dE_dx:
dE_dx = (E_inner-E_outer)/(800*(pos3(2)-pos1(2)));
% Find pressure gradient:
dp_dx = rho_0*g*dE_dx;
% Find acceleration term: -(1/rho_0)*dp_dx
p_acc = dp_dx/rho_0;

figure, plot(t, [p_acc V_acc])
figure, plot(t, windV)
%%

figure, plot(t, [p_acc-V_acc 0.0001*U])
%%
figure, subplot(1,2,1), plot(t, [E2 - E1])
subplot(1,2,2), plot(t, V)

%%
figure, plot(V_acc, (E_inner-E_outer),'.')

%%
figure,plot(V,E_inner)
%%
figure, plot(t, [V_acc 5*(E_inner-E_outer)],'-')

File: Python code/pressureDrivenCurrent.py
from netCDF4 import Dataset
from getVariable import get_variable
import numpy as np
import matplotlib.pyplot as plt

filename = 'dataset.nc'
ncid = Dataset(filename, 'r')

# Define positions
pos1 = (176, 53)
pos2 = (176, 56)
pos3 = (176, 59)
pos1 = (200, 247)
pos2 = (200, 250)
pos3 = (200, 253)


# Extracting data
t, E_inner = get_variable(filename, 'elevation', pos1)
t, E_outer = get_variable(filename, 'elevation', pos3)
t, V = get_variable(filename, 'v-velocity', pos2, 10)
t, U = get_variable(filename, 'u-velocity', pos2, 10)
t, T = get_variable(filename, 'temperature', pos2, 10)
t, S = get_variable(filename, 'salinity', pos2, 10)
t, windV = get_variable(filename, 'v-Wind', pos2)

# Calculate V acceleration
V_acc = np.diff(V) / (86400 * (t[1] - t[0]))

ncid.close()

g = 9.81
rho_0 = 1023.6  # Density of seawater

# Calculating dE_dx and pressure gradient
dE_dx = (E_inner - E_outer) / (800 * (pos3[1] - pos1[1]))
dp_dx = rho_0 * g * dE_dx

# Calculating acceleration term
p_acc = dp_dx / rho_0

# Plotting
plt.figure()
plt.plot(t, p_acc, label='p_acc')
plt.plot(t[1:], V_acc, label='V_acc')
plt.legend()
plt.show()

plt.figure()
plt.plot(t, windV)
plt.show()

# Additional plots can be created following the same pattern
